### PR TITLE
Add handlers for location as NavSatFixes to/from ROS

### PR DESCRIPTION
**Pull request by LoyVanBeek**
Send and receive locations in Telegram as NavSatFix in ROS. 
NavSatFix seemed the most relevant and best message type for this, because it is after all a NavSatFix if you share your location through Telegram base don your phone's GPS. 

Another benefit of NavSatFix is that this does not add more dependencies

### CODE REVIEW

**Comment by Timple** [comment]
Haha, nice!

File: scripts/telegram_ros_bridge [hunk 1]
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
...
+                navsatfix.position_covariance_type = NavSatFix.COVARIANCE_TYPE_UNKNOWN
+                self._from_telegram_location_publisher.publish(navsatfix)
+
+    def _ros_location_callback(self, msg):
+        """
+        Called when a new ROS NavSatFix message is coming in that should be send to the telegram conversation
+        :param msg: NavSatFix that the robot wants to share
+        """
+        if not self._telegram_chat_id:
+            rospy.logerr("ROS Bridge not initialized, dropping string message")
```
Comment (MatthijsBurgh) [inline scripts/telegram_ros_bridge]: String message, copy paste error

File: scripts/telegram_ros_bridge [hunk 2]
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
             self._telegram_updater.bot.send_photo(self._telegram_chat_id,
                                                   photo=StringIO(cv2.imencode('.jpg', cv2_img)[1].tostring()))
 
+    def _telegram_location_callback(self, _, update):
+        """
+        Called when a new telegram Location is received. The method will verify whether the incoming Location is
+        from the bridges telegram conversation by comparing the chat_id.
```
Comment (reinzor) [inline scripts/telegram_ros_bridge]: typo, bridges --> bridged

File: scripts/telegram_ros_bridge [hunk 3]
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
...
+        :param update: Received update that holds the chat_id and message data
+        """
+        location = update.message.location
+        if isinstance(location, Location):
```
Comment (reinzor) [inline scripts/telegram_ros_bridge]: Why this check, log if we receive something else?

File: scripts/telegram_ros_bridge [hunk 4]
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
...
+            rospy.loginfo("Incoming location from telegram: {}".format(location))
+            if self._telegram_chat_id is None:
+                rospy.logwarn("Discarding message. No active chat_id.")
```
Comment (reinzor) [inline scripts/telegram_ros_bridge]: Duplicated code .. Maybe we should refactor this in another PR

File: scripts/telegram_ros_bridge [hunk 5]
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
...
+                update.message.reply_text("ROS Bridge not initialized. Type /start to set-up ROS bridge")
+            elif self._telegram_chat_id != update.message.chat_id:
+                rospy.logwarn("Discarding message. Invalid chat_id")
+                update.message.reply_text(
+                    "ROS Bridge initialized to another chat_id. Type /start to connect to this chat_id")
+            else:
+                navsatfix = NavSatFix()
+                navsatfix.header.stamp = update.message.date # TODO convert tot ROS Time
```
Comment (reinzor) [inline scripts/telegram_ros_bridge]: Can be removed I guess

File: scripts/telegram_ros_bridge [hunk 6]
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
...
+                sec_since_epoch = (update.message.date - datetime.datetime(1970,1,1)).total_seconds()
+                navsatfix.header.stamp = rospy.Time.from_sec(sec_since_epoch)
+                navsatfix.header.frame_id = "telegram"
```
Comment (reinzor) [inline scripts/telegram_ros_bridge]: "frame_id is the frame of reference reported by the satellite, usually the location of the antenna.", this does not make sense. Better leave it empty if unknown.

**Review by reinzor — CHANGES_REQUESTED** [review]

**Comment by reinzor** [comment]
Cool PR :)

File: scripts/telegram_ros_bridge (hunk 5 above)
Comment (LoyVanBeek) [reply scripts/telegram_ros_bridge]: Hey, I thought I fixed this. I do want to include the timestamp the user sent the location

File: scripts/telegram_ros_bridge (hunk 4 above)
Comment (LoyVanBeek) [reply scripts/telegram_ros_bridge]: https://github.com/tue-robotics/telegram_ros/issues/2

File: scripts/telegram_ros_bridge (hunk 5 above)
Comment (LoyVanBeek) [reply scripts/telegram_ros_bridge]: Ah, I see now, done correctly in the lines below. 

I made all this on the train with laptop battery dying so i had to be quick :-)

File: scripts/telegram_ros_bridge (hunk 3 above)
Comment (LoyVanBeek) [reply scripts/telegram_ros_bridge]: I removed the check, it's useless now

**Review by reinzor — APPROVED** [review]
PR seems fine to me. Did not test this so I assume the functionality is working.

**Comment by LoyVanBeek** [comment]
I did of course test this, works quite nicely. We're going to use this to share the location of a little autonomous boat :-)

**Pull request merged**